libraries/SITL/SIM_SilentWings.cpp

```diff
@@ -209,16 +212,19 @@ void SilentWings::process_packet()
         printf("SilentWings home reset dist=%f alt=%.1f/%.1f\n",
                curr_location.get_distance(location), curr_location.alt*0.01f, location.alt*0.01f);
         // reset home location
-        home.lat = curr_location.lat;
-        home.lng = curr_location.lng;
-        origin.lat = home.lat;
-        origin.lng = home.lng;
         // Resetting altitude reference point in flight can throw off a bunch
         // of important calculations, so let the home altitude always be 0m MSL
-        home.alt = 0;
+        home = {
+            curr_location.lat,
+            curr_location.lng,
+            0,
+            Location::AltFrame::ABSOLUTE,
+        };
+        origin.lat = home.lat;
+        origin.lng = home.lng;
         position.x = 0;
         position.y = 0;
-        position.z = -curr_location.alt;
+        position.z = -alt_abs_m * 100;;
```

Comment:
looks wrong, maybe existing bug? position vector is used in meters in other places

Reply:
Yes, I agree, it does look suspect!  Not here to fix bugs ~~, just create them~~

Reply:
... well, we did just end up changing this blind